paltas/Sampling/distributions.py

```diff
@@ -355,8 +355,9 @@ def __call__(self):
 		z_lens = self.z_lens_dist()
 		clip = (z_lens - self.z_source_mean) / self.z_source_std
 		# number of std. devs away to stop (negative)
-		if(clip > self.z_source_min):
+		if(clip > self.z_source_min_default):
 			self.z_source_min = clip
```

Comment:
z_source_min here instead of self.z_source_min would be a cleaner fix. In the current fix a new self.property is created during the call, which doesn't seem like optimal behavior since that property is never referenced outside the call.